### PR TITLE
Adding slices for libldap-2.5 and dependencies

### PR DESCRIPTION
**Pull request by nhart12**
Adds slices for ldap and dependencies.

### CODE REVIEW

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
this is great! thanks @nhart12 

I'll try to review it before EOW, but in the meantime (since we don't yet have any automated linters) could you please ensure your files are indented with 2 spaces? Can you please also add a new line at the end of each file?

Thanks :pray:

**Review by woky — COMMENTED** [review]
Don't like the library symlink. Otherwise, LGTM.

File: slices/libldap-2.5-0.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    essential:
+      - libc6_libs
+      - libgnutls30_libs
+      - libsasl2-2_libs
+    contents:
+      /usr/lib/*-linux-*/libldap-2.5.so.*:
+      /usr/lib/*-linux-*/liblber-2.5.so.*:
+      #symlink workaround for https://github.com/dotnet/runtime/issues/69456
+      /usr/lib/x86_64-linux-gnu/libldap-2.4.so.2:
+        { symlink: /usr/lib/x86_64-linux-gnu/libldap-2.5.so.0 }
```
[the remark this answers is not in the document]
Comment (nhart12) [reply slices/libldap-2.5-0.yaml]: agreed, had that in there for my local build and forgot to remove it for this PR

[the remark this answers is not in the document]
Comment (nhart12) [reply slices/libldap-2.5-0.yaml]: removed

**Review by woky — APPROVED** [review]

File: slices/libgmp10.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿package: libgmp10
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+    contents:
+      /usr/lib/*-linux-*/libgmp.so.*:
```
Comment (cjdcordeiro) [inline slices/libgmp10.yaml:8]:
```suggestion
      /usr/lib/*-linux-*/libgmp.so.10*:
```

File: slices/libgnutls30.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      - libc6_libs
+      - libgmp10_libs
+      - libhogweed6_libs
+      - libidn2-0_libs
+      - libnettle8_libs
+      - libp11-kit0_libs
+      - libtasn1-6_libs
+      - libunistring2_libs
+    contents:
+      /usr/lib/*-linux-*/libgnutls*.so.*:
```
Comment (cjdcordeiro) [inline slices/libgnutls30.yaml:15]:
```suggestion
      /usr/lib/*-linux-*/libgnutls*.so.30*:
```

File: slices/libhogweed6.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿package: libhogweed6
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+      - libgmp10_libs
+      - libnettle8_libs
+    contents:
+      /usr/lib/*-linux-*/libhogweed.so*:
```
Comment (cjdcordeiro) [inline slices/libhogweed6.yaml:10]:
```suggestion
      /usr/lib/*-linux-*/libhogweed.so.6*:
```

File: slices/libidn2-0.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿package: libidn2-0
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+      - libunistring2_libs
+    contents:
+      /usr/lib/*-linux-*/libidn2.so*:
```
Comment (cjdcordeiro) [inline slices/libidn2-0.yaml:9]:
```suggestion
      /usr/lib/*-linux-*/libidn2.so.0*:
```

File: slices/libldap-2.5-0.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+      - libgnutls30_libs
+      - libsasl2-2_libs
+    contents:
+      /usr/lib/*-linux-*/libldap-2.5.so.*:
+      /usr/lib/*-linux-*/liblber-2.5.so.*:
```
Comment (cjdcordeiro) [inline slices/libldap-2.5-0.yaml:11]:
```suggestion
      /usr/lib/*-linux-*/libldap-2.5.so.0*:
      /usr/lib/*-linux-*/liblber-2.5.so.0*:
```

File: slices/libnettle8.yaml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+﻿package: libnettle8
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+    contents:
+      /usr/lib/*-linux-*/libnettle.so*:
```
Comment (cjdcordeiro) [inline slices/libnettle8.yaml:8]:
```suggestion
      /usr/lib/*-linux-*/libnettle.so.8*:
```

File: slices/libp11-kit0.yaml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+﻿package: libp11-kit0
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+      - libffi8_libs
+    contents:
+      /usr/lib/*-linux-*/libp11-kit.so.*:
```
Comment (cjdcordeiro) [inline slices/libp11-kit0.yaml:9]:
```suggestion
      /usr/lib/*-linux-*/libp11-kit.so.0*:
```

File: slices/libsasl2-2.yaml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+﻿package: libsasl2-2
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+      - libsasl2-modules-db_libs
+    contents:
+      /usr/lib/*-linux-*/libsasl*.so.*:
```
Comment (cjdcordeiro) [inline slices/libsasl2-2.yaml:9]:
```suggestion
      /usr/lib/*-linux-*/libsasl*.so.2*:
```

File: slices/libsasl2-modules-db.yaml [hunk 10]
```diff
@@ -0,0 +1,9 @@
+﻿package: libsasl2-modules-db
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+      - libdb5.3_libs
+    contents:
+      /usr/lib/*-linux-*/sasl2/*:
```
Comment (cjdcordeiro) [inline slices/libsasl2-modules-db.yaml:9]:
```suggestion
      /usr/lib/*-linux-*/sasl2/libsasldb.so*:
```

File: slices/libtasn1-6.yaml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+﻿package: libtasn1-6
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+    contents:
+      /usr/lib/*-linux-*/libtasn*.so.*:
```
Comment (cjdcordeiro) [inline slices/libtasn1-6.yaml:8]:
```suggestion
      /usr/lib/*-linux-*/libtasn1.so.6*:
```

File: slices/libunistring2.yaml [hunk 12]
```diff
@@ -0,0 +1,8 @@
+﻿package: libunistring2
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+    contents:
+      /usr/lib/*-linux-*/libunistring.so*:
```
Comment (cjdcordeiro) [inline slices/libunistring2.yaml:8]:
```suggestion
      /usr/lib/*-linux-*/libunistring.so.2*:
```

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
Thanks for the updates!

I'm suggesting a few changes to make the globs a but more restrictive in order to avoid future conflicts, but after that, it LGTM ;)

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
thanks, but you've forgotten this: https://github.com/canonical/chisel-releases/pull/55#discussion_r1305373916

For future reference, it is easier to simply accept/commit the suggestions as that will create a nicer tracing story in GH. With the force push, I had to compare the suggestions manually, which is very error-prone.

**Comment by nhart12** [comment]
> thanks, but you've forgotten this: [#55 (comment)](https://github.com/canonical/chisel-releases/pull/55#discussion_r1305373916)
> 
> For future reference, it is easier to simply accept/commit the suggestions as that will create a nicer tracing story in GH. With the force push, I had to compare the suggestions manually, which is very error-prone.

yeah I noticed that after the fact, sorry about that

**Comment by nhart12** [comment]
Thank you. Just for reference I noticed this has a lot of the same libs as defined here so may need a rebase
https://github.com/canonical/chisel-releases/pull/49

**Pull request merged**